test/server/script-processor-test.js

```diff
@@ -356,115 +356,177 @@ describe('Script processor', function () {
 
     it('Should process eval()', function () {
         testProcessing([
-            { src: 'eval(script)', expected: 'eval(__proc$Script(script))' },
-            { src: 'eval("script")', expected: 'eval(__proc$Script("script"))' },
-            { src: 'window.eval(script)', expected: 'window.eval(__proc$Script(script))' },
-            { src: 'window["eval"](script)', expected: 'window["eval"](__proc$Script(script))' },
-
-            { src: 'eval.call(window, script)', expected: 'eval.call(window, __proc$Script(script))' },
-            { src: 'eval.call(window, "script")', expected: 'eval.call(window, __proc$Script("script"))' },
-            { src: 'window.eval.call(window, script)', expected: 'window.eval.call(window, __proc$Script(script))' },
+            {
+                src:      'eval(script)',
+                expected: '(eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval)(script)'
```

Comment:
@churkin According to https://html.spec.whatwg.org/multipage/webappapis.html#timer-initialisation-steps timer script context should be set to `environment settings object`, which is, if I get right, is a global scope. Can you check, please, on your side in all browsers?